### DISCUSSION TITLE
Can exclusions be skipped?

### DISCUSSION THREAD

**Discussion by durian** [GothenburgBitFactory/timewarrior#435] [Q&A] [answered]
Hej,

I have set up my 8 hours working day with a lunch break from 11:30 - 12:30. Unfortunately, sometimes there is no time for lunch, and the clock can keep on running until after the lunch break. This messes up my hours worked because the lunch hour is deducted anyway.

I try to fix them with the lengthen/fill and other options, but that seems the wrong way to do it. What is the best way to solve this?

**Comment by lauft** [comment, marked as answer]
Hi @durian!

Another way to "fix" such a gap would be using the `join` command.
Suppose you have a gap like this:
```
$ timew sum :ids

Wk  Date       Day ID Tags    Start      End    Time   Total
W25 2021-06-23 Wed @2 WORK  8:00:00 11:30:00 3:30:00
                   @1 WORK 12:30:00 16:00:00 3:30:00 7:00:00

                                                     7:00:00
```
then you can use `timew join` to merge these two intervals:
```
$ timew join @1 @2
Joined @1 and @2

$ timew sum :ids

Wk  Date       Day ID Tags   Start      End    Time   Total
W25 2021-06-23 Wed @1 WORK 8:00:00 16:00:00 8:00:00 8:00:00

                                                    8:00:00
```

**Reply by durian** [reply]
Hi,

@lauft Thanks! Yes, that looks a lot easier than my fiddling with moves and lengthens :-)
 
Still, I wonder if maybe an option to make the exclusions hard or soft is something that could work. The "soft" ones will only be used in calculating the hours-per-day/week that should be completed, but the clock will not be stopped/started automatically. The "hard" exclusions will work as they do now. Of course, the soft ones will make the automatic tracking more difficult. 

Yet another option would be to have a setting which specifies "default hours per day" or something, if no other exclusions like holidays or lunch breaks are specified.

Anyway, thanks for the "join" solution, I hadn't found that one yet in the docs!

**Reply by lauft** [reply]
A little note on exclusions in Timewarrior: exclusions do not start or stop the clock, but create so-called _virtual intervals_ which do not exist in the database (here you will only find the current open interval) until you "touch" (i.e. modify/delete) one of them, or close the current open interval. Then the exclusions are applied and the resulting intervals stored in the database.

I am not sure about the concept of soft/hard exclusions... Maybe a hint wich suppresses application of exclusions might be more feasible 🤔
